external/storm-kafka-client/src/main/java/org/apache/storm/kafka/bolt/KafkaBolt.java

```diff
@@ -129,6 +140,27 @@ public void prepare(Map<String, Object> topoConf, TopologyContext context, Outpu
         return new KafkaProducer<>(props);
     }
 
+    /**
+     * Make the producer Callback. Using this Callback will also execute the user defined Callback, if provided.
+     */
+    protected Callback mkProducerCallback(final Tuple input) {
```

Comment:
I'm not sure we want to make this protected. The default makes use of some private fields (e.g. collector), so subclasses won't be able to implement this properly, and I'm also not really understanding why someone would need to override this, since the code that is already here is pretty important to the bolt working correctly.

Reply:
I can change to `private`.

Reply:
Fair point. I'll change it to `private`.